// @patternfly/react-charts/dist/umd/components/ChartTheme/themes/axis-theme.js
(function (global, factory) {
  if (typeof define === "function" && define.amd) {
    define(["exports", "@patternfly/react-tokens/dist/js/chart_axis_grid_stroke_Color", "@patternfly/react-tokens/dist/js/chart_axis_tick_stroke_Color"], factory);
  } else if (typeof exports !== "undefined") {
    factory(exports, require("@patternfly/react-tokens/dist/js/chart_axis_grid_stroke_Color"), require("@patternfly/react-tokens/dist/js/chart_axis_tick_stroke_Color"));
  } else {
    var mod = {
      exports: {}
    };
    factory(mod.exports, global.chart_axis_grid_stroke_Color, global.chart_axis_tick_stroke_Color);
    global.undefined = mod.exports;
  }
})(this, function (exports, _chart_axis_grid_stroke_Color, _chart_axis_tick_stroke_Color) {
  "use strict";

  Object.defineProperty(exports, "__esModule", {
    value: true
  });
  exports.AxisTheme = undefined;

  var _chart_axis_grid_stroke_Color2 = _interopRequireDefault(_chart_axis_grid_stroke_Color);

  var _chart_axis_tick_stroke_Color2 = _interopRequireDefault(_chart_axis_tick_stroke_Color);

  function _interopRequireDefault(obj) {
    return obj && obj.__esModule ? obj : {
      default: obj
    };
  }

  /* eslint-disable camelcase */
  // Axis theme
  const AxisTheme = exports.AxisTheme = {
    axis: {
      style: {
        grid: {
          stroke: _chart_axis_grid_stroke_Color2.default.value
        },
        ticks: {
          stroke: _chart_axis_tick_stroke_Color2.default.value
        }
      }
    }
  };
});
