// @patternfly/react-charts/dist/umd/components/ChartTheme/themes/base-theme.js
(function (global, factory) {
  if (typeof define === "function" && define.amd) {
    define(["exports", "@patternfly/react-tokens/dist/js/chart_global_FontFamily", "@patternfly/react-tokens/dist/js/chart_global_letter_spacing", "@patternfly/react-tokens/dist/js/chart_global_FontSize_sm", "@patternfly/react-tokens/dist/js/chart_global_label_Padding", "@patternfly/react-tokens/dist/js/chart_global_label_stroke", "@patternfly/react-tokens/dist/js/chart_global_label_text_anchor", "@patternfly/react-tokens/dist/js/chart_global_layout_Padding", "@patternfly/react-tokens/dist/js/chart_global_layout_Height", "@patternfly/react-tokens/dist/js/chart_global_layout_Width", "@patternfly/react-tokens/dist/js/chart_global_stroke_line_cap", "@patternfly/react-tokens/dist/js/chart_global_stroke_line_join", "@patternfly/react-tokens/dist/js/chart_area_data_Fill", "@patternfly/react-tokens/dist/js/chart_area_Opacity", "@patternfly/react-tokens/dist/js/chart_area_stroke_Width", "@patternfly/react-tokens/dist/js/chart_axis_axis_stroke_Width", "@patternfly/react-tokens/dist/js/chart_axis_axis_stroke_Color", "@patternfly/react-tokens/dist/js/chart_axis_axis_Fill", "@patternfly/react-tokens/dist/js/chart_axis_axis_label_Padding", "@patternfly/react-tokens/dist/js/chart_axis_axis_label_stroke_Color", "@patternfly/react-tokens/dist/js/chart_axis_grid_Fill", "@patternfly/react-tokens/dist/js/chart_axis_grid_PointerEvents", "@patternfly/react-tokens/dist/js/chart_axis_tick_Fill", "@patternfly/react-tokens/dist/js/chart_axis_tick_Size", "@patternfly/react-tokens/dist/js/chart_axis_tick_stroke_Color", "@patternfly/react-tokens/dist/js/chart_axis_tick_Width", "@patternfly/react-tokens/dist/js/chart_axis_tick_label_Fill", "@patternfly/react-tokens/dist/js/chart_bar_Width", "@patternfly/react-tokens/dist/js/chart_bar_data_stroke", "@patternfly/react-tokens/dist/js/chart_bar_data_Fill", "@patternfly/react-tokens/dist/js/chart_bar_data_Padding", "@patternfly/react-tokens/dist/js/chart_bar_data_stroke_Width", "@patternfly/react-tokens/dist/js/chart_boxplot_max_Padding", "@patternfly/react-tokens/dist/js/chart_boxplot_max_stroke_Color", "@patternfly/react-tokens/dist/js/chart_boxplot_max_stroke_Width", "@patternfly/react-tokens/dist/js/chart_boxplot_median_Padding", "@patternfly/react-tokens/dist/js/chart_boxplot_median_stroke_Color", "@patternfly/react-tokens/dist/js/chart_boxplot_median_stroke_Width", "@patternfly/react-tokens/dist/js/chart_boxplot_min_Padding", "@patternfly/react-tokens/dist/js/chart_boxplot_min_stroke_Width", "@patternfly/react-tokens/dist/js/chart_boxplot_min_stroke_Color", "@patternfly/react-tokens/dist/js/chart_boxplot_lower_quartile_Padding", "@patternfly/react-tokens/dist/js/chart_boxplot_lower_quartile_Fill", "@patternfly/react-tokens/dist/js/chart_boxplot_upper_quartile_Padding", "@patternfly/react-tokens/dist/js/chart_boxplot_upper_quartile_Fill", "@patternfly/react-tokens/dist/js/chart_boxplot_box_Width", "@patternfly/react-tokens/dist/js/chart_candelstick_data_stroke_Width", "@patternfly/react-tokens/dist/js/chart_candelstick_data_stroke_Color", "@patternfly/react-tokens/dist/js/chart_candelstick_candle_positive_Color", "@patternfly/react-tokens/dist/js/chart_candelstick_candle_negative_Color", "@patternfly/react-tokens/dist/js/chart_errorbar_BorderWidth", "@patternfly/react-tokens/dist/js/chart_errorbar_data_Fill", "@patternfly/react-tokens/dist/js/chart_errorbar_data_Opacity", "@patternfly/react-tokens/dist/js/chart_errorbar_data_stroke_Width", "@patternfly/react-tokens/dist/js/chart_errorbar_data_stroke_Color", "@patternfly/react-tokens/dist/js/chart_legend_gutter_Width", "@patternfly/react-tokens/dist/js/chart_legend_orientation", "@patternfly/react-tokens/dist/js/chart_legend_title_orientation", "@patternfly/react-tokens/dist/js/chart_legend_data_type", "@patternfly/react-tokens/dist/js/chart_legend_title_Padding", "@patternfly/react-tokens/dist/js/chart_line_data_Fill", "@patternfly/react-tokens/dist/js/chart_line_data_Opacity", "@patternfly/react-tokens/dist/js/chart_line_data_stroke_Width", "@patternfly/react-tokens/dist/js/chart_line_data_stroke_Color", "@patternfly/react-tokens/dist/js/chart_pie_Padding", "@patternfly/react-tokens/dist/js/chart_pie_data_Padding", "@patternfly/react-tokens/dist/js/chart_pie_data_stroke_Width", "@patternfly/react-tokens/dist/js/chart_pie_data_stroke_Color", "@patternfly/react-tokens/dist/js/chart_pie_labels_Padding", "@patternfly/react-tokens/dist/js/chart_pie_Height", "@patternfly/react-tokens/dist/js/chart_pie_Width", "@patternfly/react-tokens/dist/js/chart_scatter_data_stroke_Color", "@patternfly/react-tokens/dist/js/chart_scatter_data_stroke_Width", "@patternfly/react-tokens/dist/js/chart_scatter_data_Opacity", "@patternfly/react-tokens/dist/js/chart_scatter_data_Fill", "@patternfly/react-tokens/dist/js/chart_stack_data_stroke_Width", "@patternfly/react-tokens/dist/js/chart_tooltip_corner_radius", "@patternfly/react-tokens/dist/js/chart_tooltip_pointer_length", "@patternfly/react-tokens/dist/js/chart_tooltip_Fill", "@patternfly/react-tokens/dist/js/chart_tooltip_flyoutStyle_corner_radius", "@patternfly/react-tokens/dist/js/chart_tooltip_flyoutStyle_stroke_Width", "@patternfly/react-tokens/dist/js/chart_tooltip_flyoutStyle_PointerEvents", "@patternfly/react-tokens/dist/js/chart_tooltip_flyoutStyle_stroke_Color", "@patternfly/react-tokens/dist/js/chart_tooltip_flyoutStyle_Fill", "@patternfly/react-tokens/dist/js/chart_tooltip_pointer_Width", "@patternfly/react-tokens/dist/js/chart_tooltip_Padding", "@patternfly/react-tokens/dist/js/chart_tooltip_PointerEvents", "@patternfly/react-tokens/dist/js/chart_voronoi_data_Fill", "@patternfly/react-tokens/dist/js/chart_voronoi_data_stroke_Color", "@patternfly/react-tokens/dist/js/chart_voronoi_data_stroke_Width", "@patternfly/react-tokens/dist/js/chart_voronoi_labels_Fill", "@patternfly/react-tokens/dist/js/chart_voronoi_labels_Padding", "@patternfly/react-tokens/dist/js/chart_voronoi_labels_PointerEvents", "@patternfly/react-tokens/dist/js/chart_voronoi_flyout_stroke_Width", "@patternfly/react-tokens/dist/js/chart_voronoi_flyout_PointerEvents", "@patternfly/react-tokens/dist/js/chart_voronoi_flyout_stroke_Color", "@patternfly/react-tokens/dist/js/chart_voronoi_flyout_stroke_Fill"], factory);
  } else if (typeof exports !== "undefined") {
    factory(exports, require("@patternfly/react-tokens/dist/js/chart_global_FontFamily"), require("@patternfly/react-tokens/dist/js/chart_global_letter_spacing"), require("@patternfly/react-tokens/dist/js/chart_global_FontSize_sm"), require("@patternfly/react-tokens/dist/js/chart_global_label_Padding"), require("@patternfly/react-tokens/dist/js/chart_global_label_stroke"), require("@patternfly/react-tokens/dist/js/chart_global_label_text_anchor"), require("@patternfly/react-tokens/dist/js/chart_global_layout_Padding"), require("@patternfly/react-tokens/dist/js/chart_global_layout_Height"), require("@patternfly/react-tokens/dist/js/chart_global_layout_Width"), require("@patternfly/react-tokens/dist/js/chart_global_stroke_line_cap"), require("@patternfly/react-tokens/dist/js/chart_global_stroke_line_join"), require("@patternfly/react-tokens/dist/js/chart_area_data_Fill"), require("@patternfly/react-tokens/dist/js/chart_area_Opacity"), require("@patternfly/react-tokens/dist/js/chart_area_stroke_Width"), require("@patternfly/react-tokens/dist/js/chart_axis_axis_stroke_Width"), require("@patternfly/react-tokens/dist/js/chart_axis_axis_stroke_Color"), require("@patternfly/react-tokens/dist/js/chart_axis_axis_Fill"), require("@patternfly/react-tokens/dist/js/chart_axis_axis_label_Padding"), require("@patternfly/react-tokens/dist/js/chart_axis_axis_label_stroke_Color"), require("@patternfly/react-tokens/dist/js/chart_axis_grid_Fill"), require("@patternfly/react-tokens/dist/js/chart_axis_grid_PointerEvents"), require("@patternfly/react-tokens/dist/js/chart_axis_tick_Fill"), require("@patternfly/react-tokens/dist/js/chart_axis_tick_Size"), require("@patternfly/react-tokens/dist/js/chart_axis_tick_stroke_Color"), require("@patternfly/react-tokens/dist/js/chart_axis_tick_Width"), require("@patternfly/react-tokens/dist/js/chart_axis_tick_label_Fill"), require("@patternfly/react-tokens/dist/js/chart_bar_Width"), require("@patternfly/react-tokens/dist/js/chart_bar_data_stroke"), require("@patternfly/react-tokens/dist/js/chart_bar_data_Fill"), require("@patternfly/react-tokens/dist/js/chart_bar_data_Padding"), require("@patternfly/react-tokens/dist/js/chart_bar_data_stroke_Width"), require("@patternfly/react-tokens/dist/js/chart_boxplot_max_Padding"), require("@patternfly/react-tokens/dist/js/chart_boxplot_max_stroke_Color"), require("@patternfly/react-tokens/dist/js/chart_boxplot_max_stroke_Width"), require("@patternfly/react-tokens/dist/js/chart_boxplot_median_Padding"), require("@patternfly/react-tokens/dist/js/chart_boxplot_median_stroke_Color"), require("@patternfly/react-tokens/dist/js/chart_boxplot_median_stroke_Width"), require("@patternfly/react-tokens/dist/js/chart_boxplot_min_Padding"), require("@patternfly/react-tokens/dist/js/chart_boxplot_min_stroke_Width"), require("@patternfly/react-tokens/dist/js/chart_boxplot_min_stroke_Color"), require("@patternfly/react-tokens/dist/js/chart_boxplot_lower_quartile_Padding"), require("@patternfly/react-tokens/dist/js/chart_boxplot_lower_quartile_Fill"), require("@patternfly/react-tokens/dist/js/chart_boxplot_upper_quartile_Padding"), require("@patternfly/react-tokens/dist/js/chart_boxplot_upper_quartile_Fill"), require("@patternfly/react-tokens/dist/js/chart_boxplot_box_Width"), require("@patternfly/react-tokens/dist/js/chart_candelstick_data_stroke_Width"), require("@patternfly/react-tokens/dist/js/chart_candelstick_data_stroke_Color"), require("@patternfly/react-tokens/dist/js/chart_candelstick_candle_positive_Color"), require("@patternfly/react-tokens/dist/js/chart_candelstick_candle_negative_Color"), require("@patternfly/react-tokens/dist/js/chart_errorbar_BorderWidth"), require("@patternfly/react-tokens/dist/js/chart_errorbar_data_Fill"), require("@patternfly/react-tokens/dist/js/chart_errorbar_data_Opacity"), require("@patternfly/react-tokens/dist/js/chart_errorbar_data_stroke_Width"), require("@patternfly/react-tokens/dist/js/chart_errorbar_data_stroke_Color"), require("@patternfly/react-tokens/dist/js/chart_legend_gutter_Width"), require("@patternfly/react-tokens/dist/js/chart_legend_orientation"), require("@patternfly/react-tokens/dist/js/chart_legend_title_orientation"), require("@patternfly/react-tokens/dist/js/chart_legend_data_type"), require("@patternfly/react-tokens/dist/js/chart_legend_title_Padding"), require("@patternfly/react-tokens/dist/js/chart_line_data_Fill"), require("@patternfly/react-tokens/dist/js/chart_line_data_Opacity"), require("@patternfly/react-tokens/dist/js/chart_line_data_stroke_Width"), require("@patternfly/react-tokens/dist/js/chart_line_data_stroke_Color"), require("@patternfly/react-tokens/dist/js/chart_pie_Padding"), require("@patternfly/react-tokens/dist/js/chart_pie_data_Padding"), require("@patternfly/react-tokens/dist/js/chart_pie_data_stroke_Width"), require("@patternfly/react-tokens/dist/js/chart_pie_data_stroke_Color"), require("@patternfly/react-tokens/dist/js/chart_pie_labels_Padding"), require("@patternfly/react-tokens/dist/js/chart_pie_Height"), require("@patternfly/react-tokens/dist/js/chart_pie_Width"), require("@patternfly/react-tokens/dist/js/chart_scatter_data_stroke_Color"), require("@patternfly/react-tokens/dist/js/chart_scatter_data_stroke_Width"), require("@patternfly/react-tokens/dist/js/chart_scatter_data_Opacity"), require("@patternfly/react-tokens/dist/js/chart_scatter_data_Fill"), require("@patternfly/react-tokens/dist/js/chart_stack_data_stroke_Width"), require("@patternfly/react-tokens/dist/js/chart_tooltip_corner_radius"), require("@patternfly/react-tokens/dist/js/chart_tooltip_pointer_length"), require("@patternfly/react-tokens/dist/js/chart_tooltip_Fill"), require("@patternfly/react-tokens/dist/js/chart_tooltip_flyoutStyle_corner_radius"), require("@patternfly/react-tokens/dist/js/chart_tooltip_flyoutStyle_stroke_Width"), require("@patternfly/react-tokens/dist/js/chart_tooltip_flyoutStyle_PointerEvents"), require("@patternfly/react-tokens/dist/js/chart_tooltip_flyoutStyle_stroke_Color"), require("@patternfly/react-tokens/dist/js/chart_tooltip_flyoutStyle_Fill"), require("@patternfly/react-tokens/dist/js/chart_tooltip_pointer_Width"), require("@patternfly/react-tokens/dist/js/chart_tooltip_Padding"), require("@patternfly/react-tokens/dist/js/chart_tooltip_PointerEvents"), require("@patternfly/react-tokens/dist/js/chart_voronoi_data_Fill"), require("@patternfly/react-tokens/dist/js/chart_voronoi_data_stroke_Color"), require("@patternfly/react-tokens/dist/js/chart_voronoi_data_stroke_Width"), require("@patternfly/react-tokens/dist/js/chart_voronoi_labels_Fill"), require("@patternfly/react-tokens/dist/js/chart_voronoi_labels_Padding"), require("@patternfly/react-tokens/dist/js/chart_voronoi_labels_PointerEvents"), require("@patternfly/react-tokens/dist/js/chart_voronoi_flyout_stroke_Width"), require("@patternfly/react-tokens/dist/js/chart_voronoi_flyout_PointerEvents"), require("@patternfly/react-tokens/dist/js/chart_voronoi_flyout_stroke_Color"), require("@patternfly/react-tokens/dist/js/chart_voronoi_flyout_stroke_Fill"));
  } else {
    var mod = {
      exports: {}
    };
    factory(mod.exports, global.chart_global_FontFamily, global.chart_global_letter_spacing, global.chart_global_FontSize_sm, global.chart_global_label_Padding, global.chart_global_label_stroke, global.chart_global_label_text_anchor, global.chart_global_layout_Padding, global.chart_global_layout_Height, global.chart_global_layout_Width, global.chart_global_stroke_line_cap, global.chart_global_stroke_line_join, global.chart_area_data_Fill, global.chart_area_Opacity, global.chart_area_stroke_Width, global.chart_axis_axis_stroke_Width, global.chart_axis_axis_stroke_Color, global.chart_axis_axis_Fill, global.chart_axis_axis_label_Padding, global.chart_axis_axis_label_stroke_Color, global.chart_axis_grid_Fill, global.chart_axis_grid_PointerEvents, global.chart_axis_tick_Fill, global.chart_axis_tick_Size, global.chart_axis_tick_stroke_Color, global.chart_axis_tick_Width, global.chart_axis_tick_label_Fill, global.chart_bar_Width, global.chart_bar_data_stroke, global.chart_bar_data_Fill, global.chart_bar_data_Padding, global.chart_bar_data_stroke_Width, global.chart_boxplot_max_Padding, global.chart_boxplot_max_stroke_Color, global.chart_boxplot_max_stroke_Width, global.chart_boxplot_median_Padding, global.chart_boxplot_median_stroke_Color, global.chart_boxplot_median_stroke_Width, global.chart_boxplot_min_Padding, global.chart_boxplot_min_stroke_Width, global.chart_boxplot_min_stroke_Color, global.chart_boxplot_lower_quartile_Padding, global.chart_boxplot_lower_quartile_Fill, global.chart_boxplot_upper_quartile_Padding, global.chart_boxplot_upper_quartile_Fill, global.chart_boxplot_box_Width, global.chart_candelstick_data_stroke_Width, global.chart_candelstick_data_stroke_Color, global.chart_candelstick_candle_positive_Color, global.chart_candelstick_candle_negative_Color, global.chart_errorbar_BorderWidth, global.chart_errorbar_data_Fill, global.chart_errorbar_data_Opacity, global.chart_errorbar_data_stroke_Width, global.chart_errorbar_data_stroke_Color, global.chart_legend_gutter_Width, global.chart_legend_orientation, global.chart_legend_title_orientation, global.chart_legend_data_type, global.chart_legend_title_Padding, global.chart_line_data_Fill, global.chart_line_data_Opacity, global.chart_line_data_stroke_Width, global.chart_line_data_stroke_Color, global.chart_pie_Padding, global.chart_pie_data_Padding, global.chart_pie_data_stroke_Width, global.chart_pie_data_stroke_Color, global.chart_pie_labels_Padding, global.chart_pie_Height, global.chart_pie_Width, global.chart_scatter_data_stroke_Color, global.chart_scatter_data_stroke_Width, global.chart_scatter_data_Opacity, global.chart_scatter_data_Fill, global.chart_stack_data_stroke_Width, global.chart_tooltip_corner_radius, global.chart_tooltip_pointer_length, global.chart_tooltip_Fill, global.chart_tooltip_flyoutStyle_corner_radius, global.chart_tooltip_flyoutStyle_stroke_Width, global.chart_tooltip_flyoutStyle_PointerEvents, global.chart_tooltip_flyoutStyle_stroke_Color, global.chart_tooltip_flyoutStyle_Fill, global.chart_tooltip_pointer_Width, global.chart_tooltip_Padding, global.chart_tooltip_PointerEvents, global.chart_voronoi_data_Fill, global.chart_voronoi_data_stroke_Color, global.chart_voronoi_data_stroke_Width, global.chart_voronoi_labels_Fill, global.chart_voronoi_labels_Padding, global.chart_voronoi_labels_PointerEvents, global.chart_voronoi_flyout_stroke_Width, global.chart_voronoi_flyout_PointerEvents, global.chart_voronoi_flyout_stroke_Color, global.chart_voronoi_flyout_stroke_Fill);
    global.undefined = mod.exports;
  }
})(this, function (exports, _chart_global_FontFamily, _chart_global_letter_spacing, _chart_global_FontSize_sm, _chart_global_label_Padding, _chart_global_label_stroke, _chart_global_label_text_anchor, _chart_global_layout_Padding, _chart_global_layout_Height, _chart_global_layout_Width, _chart_global_stroke_line_cap, _chart_global_stroke_line_join, _chart_area_data_Fill, _chart_area_Opacity, _chart_area_stroke_Width, _chart_axis_axis_stroke_Width, _chart_axis_axis_stroke_Color, _chart_axis_axis_Fill, _chart_axis_axis_label_Padding, _chart_axis_axis_label_stroke_Color, _chart_axis_grid_Fill, _chart_axis_grid_PointerEvents, _chart_axis_tick_Fill, _chart_axis_tick_Size, _chart_axis_tick_stroke_Color, _chart_axis_tick_Width, _chart_axis_tick_label_Fill, _chart_bar_Width, _chart_bar_data_stroke, _chart_bar_data_Fill, _chart_bar_data_Padding, _chart_bar_data_stroke_Width, _chart_boxplot_max_Padding, _chart_boxplot_max_stroke_Color, _chart_boxplot_max_stroke_Width, _chart_boxplot_median_Padding, _chart_boxplot_median_stroke_Color, _chart_boxplot_median_stroke_Width, _chart_boxplot_min_Padding, _chart_boxplot_min_stroke_Width, _chart_boxplot_min_stroke_Color, _chart_boxplot_lower_quartile_Padding, _chart_boxplot_lower_quartile_Fill, _chart_boxplot_upper_quartile_Padding, _chart_boxplot_upper_quartile_Fill, _chart_boxplot_box_Width, _chart_candelstick_data_stroke_Width, _chart_candelstick_data_stroke_Color, _chart_candelstick_candle_positive_Color, _chart_candelstick_candle_negative_Color, _chart_errorbar_BorderWidth, _chart_errorbar_data_Fill, _chart_errorbar_data_Opacity, _chart_errorbar_data_stroke_Width, _chart_errorbar_data_stroke_Color, _chart_legend_gutter_Width, _chart_legend_orientation, _chart_legend_title_orientation, _chart_legend_data_type, _chart_legend_title_Padding, _chart_line_data_Fill, _chart_line_data_Opacity, _chart_line_data_stroke_Width, _chart_line_data_stroke_Color, _chart_pie_Padding, _chart_pie_data_Padding, _chart_pie_data_stroke_Width, _chart_pie_data_stroke_Color, _chart_pie_labels_Padding, _chart_pie_Height, _chart_pie_Width, _chart_scatter_data_stroke_Color, _chart_scatter_data_stroke_Width, _chart_scatter_data_Opacity, _chart_scatter_data_Fill, _chart_stack_data_stroke_Width, _chart_tooltip_corner_radius, _chart_tooltip_pointer_length, _chart_tooltip_Fill, _chart_tooltip_flyoutStyle_corner_radius, _chart_tooltip_flyoutStyle_stroke_Width, _chart_tooltip_flyoutStyle_PointerEvents, _chart_tooltip_flyoutStyle_stroke_Color, _chart_tooltip_flyoutStyle_Fill, _chart_tooltip_pointer_Width, _chart_tooltip_Padding, _chart_tooltip_PointerEvents, _chart_voronoi_data_Fill, _chart_voronoi_data_stroke_Color, _chart_voronoi_data_stroke_Width, _chart_voronoi_labels_Fill, _chart_voronoi_labels_Padding, _chart_voronoi_labels_PointerEvents, _chart_voronoi_flyout_stroke_Width, _chart_voronoi_flyout_PointerEvents, _chart_voronoi_flyout_stroke_Color, _chart_voronoi_flyout_stroke_Fill) {
  "use strict";

  Object.defineProperty(exports, "__esModule", {
    value: true
  });
  exports.BaseTheme = undefined;

  var _chart_global_FontFamily2 = _interopRequireDefault(_chart_global_FontFamily);

  var _chart_global_letter_spacing2 = _interopRequireDefault(_chart_global_letter_spacing);

  var _chart_global_FontSize_sm2 = _interopRequireDefault(_chart_global_FontSize_sm);

  var _chart_global_label_Padding2 = _interopRequireDefault(_chart_global_label_Padding);

  var _chart_global_label_stroke2 = _interopRequireDefault(_chart_global_label_stroke);

  var _chart_global_label_text_anchor2 = _interopRequireDefault(_chart_global_label_text_anchor);

  var _chart_global_layout_Padding2 = _interopRequireDefault(_chart_global_layout_Padding);

  var _chart_global_layout_Height2 = _interopRequireDefault(_chart_global_layout_Height);

  var _chart_global_layout_Width2 = _interopRequireDefault(_chart_global_layout_Width);

  var _chart_global_stroke_line_cap2 = _interopRequireDefault(_chart_global_stroke_line_cap);

  var _chart_global_stroke_line_join2 = _interopRequireDefault(_chart_global_stroke_line_join);

  var _chart_area_data_Fill2 = _interopRequireDefault(_chart_area_data_Fill);

  var _chart_area_Opacity2 = _interopRequireDefault(_chart_area_Opacity);

  var _chart_area_stroke_Width2 = _interopRequireDefault(_chart_area_stroke_Width);

  var _chart_axis_axis_stroke_Width2 = _interopRequireDefault(_chart_axis_axis_stroke_Width);

  var _chart_axis_axis_stroke_Color2 = _interopRequireDefault(_chart_axis_axis_stroke_Color);

  var _chart_axis_axis_Fill2 = _interopRequireDefault(_chart_axis_axis_Fill);

  var _chart_axis_axis_label_Padding2 = _interopRequireDefault(_chart_axis_axis_label_Padding);

  var _chart_axis_axis_label_stroke_Color2 = _interopRequireDefault(_chart_axis_axis_label_stroke_Color);

  var _chart_axis_grid_Fill2 = _interopRequireDefault(_chart_axis_grid_Fill);

  var _chart_axis_grid_PointerEvents2 = _interopRequireDefault(_chart_axis_grid_PointerEvents);

  var _chart_axis_tick_Fill2 = _interopRequireDefault(_chart_axis_tick_Fill);

  var _chart_axis_tick_Size2 = _interopRequireDefault(_chart_axis_tick_Size);

  var _chart_axis_tick_stroke_Color2 = _interopRequireDefault(_chart_axis_tick_stroke_Color);

  var _chart_axis_tick_Width2 = _interopRequireDefault(_chart_axis_tick_Width);

  var _chart_axis_tick_label_Fill2 = _interopRequireDefault(_chart_axis_tick_label_Fill);

  var _chart_bar_Width2 = _interopRequireDefault(_chart_bar_Width);

  var _chart_bar_data_stroke2 = _interopRequireDefault(_chart_bar_data_stroke);

  var _chart_bar_data_Fill2 = _interopRequireDefault(_chart_bar_data_Fill);

  var _chart_bar_data_Padding2 = _interopRequireDefault(_chart_bar_data_Padding);

  var _chart_bar_data_stroke_Width2 = _interopRequireDefault(_chart_bar_data_stroke_Width);

  var _chart_boxplot_max_Padding2 = _interopRequireDefault(_chart_boxplot_max_Padding);

  var _chart_boxplot_max_stroke_Color2 = _interopRequireDefault(_chart_boxplot_max_stroke_Color);

  var _chart_boxplot_max_stroke_Width2 = _interopRequireDefault(_chart_boxplot_max_stroke_Width);

  var _chart_boxplot_median_Padding2 = _interopRequireDefault(_chart_boxplot_median_Padding);

  var _chart_boxplot_median_stroke_Color2 = _interopRequireDefault(_chart_boxplot_median_stroke_Color);

  var _chart_boxplot_median_stroke_Width2 = _interopRequireDefault(_chart_boxplot_median_stroke_Width);

  var _chart_boxplot_min_Padding2 = _interopRequireDefault(_chart_boxplot_min_Padding);

  var _chart_boxplot_min_stroke_Width2 = _interopRequireDefault(_chart_boxplot_min_stroke_Width);

  var _chart_boxplot_min_stroke_Color2 = _interopRequireDefault(_chart_boxplot_min_stroke_Color);

  var _chart_boxplot_lower_quartile_Padding2 = _interopRequireDefault(_chart_boxplot_lower_quartile_Padding);

  var _chart_boxplot_lower_quartile_Fill2 = _interopRequireDefault(_chart_boxplot_lower_quartile_Fill);

  var _chart_boxplot_upper_quartile_Padding2 = _interopRequireDefault(_chart_boxplot_upper_quartile_Padding);

  var _chart_boxplot_upper_quartile_Fill2 = _interopRequireDefault(_chart_boxplot_upper_quartile_Fill);

  var _chart_boxplot_box_Width2 = _interopRequireDefault(_chart_boxplot_box_Width);

  var _chart_candelstick_data_stroke_Width2 = _interopRequireDefault(_chart_candelstick_data_stroke_Width);

  var _chart_candelstick_data_stroke_Color2 = _interopRequireDefault(_chart_candelstick_data_stroke_Color);

  var _chart_candelstick_candle_positive_Color2 = _interopRequireDefault(_chart_candelstick_candle_positive_Color);

  var _chart_candelstick_candle_negative_Color2 = _interopRequireDefault(_chart_candelstick_candle_negative_Color);

  var _chart_errorbar_BorderWidth2 = _interopRequireDefault(_chart_errorbar_BorderWidth);

  var _chart_errorbar_data_Fill2 = _interopRequireDefault(_chart_errorbar_data_Fill);

  var _chart_errorbar_data_Opacity2 = _interopRequireDefault(_chart_errorbar_data_Opacity);

  var _chart_errorbar_data_stroke_Width2 = _interopRequireDefault(_chart_errorbar_data_stroke_Width);

  var _chart_errorbar_data_stroke_Color2 = _interopRequireDefault(_chart_errorbar_data_stroke_Color);

  var _chart_legend_gutter_Width2 = _interopRequireDefault(_chart_legend_gutter_Width);

  var _chart_legend_orientation2 = _interopRequireDefault(_chart_legend_orientation);

  var _chart_legend_title_orientation2 = _interopRequireDefault(_chart_legend_title_orientation);

  var _chart_legend_data_type2 = _interopRequireDefault(_chart_legend_data_type);

  var _chart_legend_title_Padding2 = _interopRequireDefault(_chart_legend_title_Padding);

  var _chart_line_data_Fill2 = _interopRequireDefault(_chart_line_data_Fill);

  var _chart_line_data_Opacity2 = _interopRequireDefault(_chart_line_data_Opacity);

  var _chart_line_data_stroke_Width2 = _interopRequireDefault(_chart_line_data_stroke_Width);

  var _chart_line_data_stroke_Color2 = _interopRequireDefault(_chart_line_data_stroke_Color);

  var _chart_pie_Padding2 = _interopRequireDefault(_chart_pie_Padding);

  var _chart_pie_data_Padding2 = _interopRequireDefault(_chart_pie_data_Padding);

  var _chart_pie_data_stroke_Width2 = _interopRequireDefault(_chart_pie_data_stroke_Width);

  var _chart_pie_data_stroke_Color2 = _interopRequireDefault(_chart_pie_data_stroke_Color);

  var _chart_pie_labels_Padding2 = _interopRequireDefault(_chart_pie_labels_Padding);

  var _chart_pie_Height2 = _interopRequireDefault(_chart_pie_Height);

  var _chart_pie_Width2 = _interopRequireDefault(_chart_pie_Width);

  var _chart_scatter_data_stroke_Color2 = _interopRequireDefault(_chart_scatter_data_stroke_Color);

  var _chart_scatter_data_stroke_Width2 = _interopRequireDefault(_chart_scatter_data_stroke_Width);

  var _chart_scatter_data_Opacity2 = _interopRequireDefault(_chart_scatter_data_Opacity);

  var _chart_scatter_data_Fill2 = _interopRequireDefault(_chart_scatter_data_Fill);

  var _chart_stack_data_stroke_Width2 = _interopRequireDefault(_chart_stack_data_stroke_Width);

  var _chart_tooltip_corner_radius2 = _interopRequireDefault(_chart_tooltip_corner_radius);

  var _chart_tooltip_pointer_length2 = _interopRequireDefault(_chart_tooltip_pointer_length);

  var _chart_tooltip_Fill2 = _interopRequireDefault(_chart_tooltip_Fill);

  var _chart_tooltip_flyoutStyle_corner_radius2 = _interopRequireDefault(_chart_tooltip_flyoutStyle_corner_radius);

  var _chart_tooltip_flyoutStyle_stroke_Width2 = _interopRequireDefault(_chart_tooltip_flyoutStyle_stroke_Width);

  var _chart_tooltip_flyoutStyle_PointerEvents2 = _interopRequireDefault(_chart_tooltip_flyoutStyle_PointerEvents);

  var _chart_tooltip_flyoutStyle_stroke_Color2 = _interopRequireDefault(_chart_tooltip_flyoutStyle_stroke_Color);

  var _chart_tooltip_flyoutStyle_Fill2 = _interopRequireDefault(_chart_tooltip_flyoutStyle_Fill);

  var _chart_tooltip_pointer_Width2 = _interopRequireDefault(_chart_tooltip_pointer_Width);

  var _chart_tooltip_Padding2 = _interopRequireDefault(_chart_tooltip_Padding);

  var _chart_tooltip_PointerEvents2 = _interopRequireDefault(_chart_tooltip_PointerEvents);

  var _chart_voronoi_data_Fill2 = _interopRequireDefault(_chart_voronoi_data_Fill);

  var _chart_voronoi_data_stroke_Color2 = _interopRequireDefault(_chart_voronoi_data_stroke_Color);

  var _chart_voronoi_data_stroke_Width2 = _interopRequireDefault(_chart_voronoi_data_stroke_Width);

  var _chart_voronoi_labels_Fill2 = _interopRequireDefault(_chart_voronoi_labels_Fill);

  var _chart_voronoi_labels_Padding2 = _interopRequireDefault(_chart_voronoi_labels_Padding);

  var _chart_voronoi_labels_PointerEvents2 = _interopRequireDefault(_chart_voronoi_labels_PointerEvents);

  var _chart_voronoi_flyout_stroke_Width2 = _interopRequireDefault(_chart_voronoi_flyout_stroke_Width);

  var _chart_voronoi_flyout_PointerEvents2 = _interopRequireDefault(_chart_voronoi_flyout_PointerEvents);

  var _chart_voronoi_flyout_stroke_Color2 = _interopRequireDefault(_chart_voronoi_flyout_stroke_Color);

  var _chart_voronoi_flyout_stroke_Fill2 = _interopRequireDefault(_chart_voronoi_flyout_stroke_Fill);

  function _interopRequireDefault(obj) {
    return obj && obj.__esModule ? obj : {
      default: obj
    };
  }

  function ownKeys(object, enumerableOnly) {
    var keys = Object.keys(object);

    if (Object.getOwnPropertySymbols) {
      var symbols = Object.getOwnPropertySymbols(object);
      if (enumerableOnly) symbols = symbols.filter(function (sym) {
        return Object.getOwnPropertyDescriptor(object, sym).enumerable;
      });
      keys.push.apply(keys, symbols);
    }

    return keys;
  }

  function _objectSpread(target) {
    for (var i = 1; i < arguments.length; i++) {
      var source = arguments[i] != null ? arguments[i] : {};

      if (i % 2) {
        ownKeys(source, true).forEach(function (key) {
          _defineProperty(target, key, source[key]);
        });
      } else if (Object.getOwnPropertyDescriptors) {
        Object.defineProperties(target, Object.getOwnPropertyDescriptors(source));
      } else {
        ownKeys(source).forEach(function (key) {
          Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key));
        });
      }
    }

    return target;
  }

  function _defineProperty(obj, key, value) {
    if (key in obj) {
      Object.defineProperty(obj, key, {
        value: value,
        enumerable: true,
        configurable: true,
        writable: true
      });
    } else {
      obj[key] = value;
    }

    return obj;
  }
  /* eslint-disable camelcase */


  // Note: Values must be in pixles
  // Typography
  const TYPOGRAPHY_FONT_FAMILY = _chart_global_FontFamily2.default.var;
  const TYPOGRAPHY_LETTER_SPACING = _chart_global_letter_spacing2.default.var;
  const TYPOGRAPHY_FONT_SIZE = _chart_global_FontSize_sm2.default.value; // Labels

  const LABEL_PROPS = {
    fontFamily: TYPOGRAPHY_FONT_FAMILY,
    fontSize: TYPOGRAPHY_FONT_SIZE,
    letterSpacing: TYPOGRAPHY_LETTER_SPACING,
    padding: _chart_global_label_Padding2.default.value,
    stroke: _chart_global_label_stroke2.default.value
  };

  const LABEL_CENTERED_PROPS = _objectSpread({}, LABEL_PROPS, {
    textAnchor: _chart_global_label_text_anchor2.default.value
  }); // Layout


  const LAYOUT_PROPS = {
    padding: _chart_global_layout_Padding2.default.value,
    height: _chart_global_layout_Height2.default.value,
    width: _chart_global_layout_Width2.default.value
  }; // Strokes

  const STROKE_LINE_CAP = _chart_global_stroke_line_cap2.default.value;
  const STROKE_LINE_JOIN = _chart_global_stroke_line_join2.default.value; // Victory theme properties only

  const BaseTheme = exports.BaseTheme = {
    area: _objectSpread({}, LAYOUT_PROPS, {
      style: {
        data: {
          fill: _chart_area_data_Fill2.default.value,
          fillOpacity: _chart_area_Opacity2.default.value,
          // Omit stroke to add a line border from color scale
          // stroke: chart_global_label_stroke.value,
          strokeWidth: _chart_area_stroke_Width2.default.value
        },
        labels: LABEL_CENTERED_PROPS
      }
    }),
    axis: _objectSpread({}, LAYOUT_PROPS, {
      style: {
        axis: {
          fill: _chart_axis_axis_Fill2.default.value,
          strokeWidth: _chart_axis_axis_stroke_Width2.default.value,
          stroke: _chart_axis_axis_stroke_Color2.default.value,
          strokeLinecap: STROKE_LINE_CAP,
          strokeLinejoin: STROKE_LINE_JOIN
        },
        axisLabel: _objectSpread({}, LABEL_CENTERED_PROPS, {
          padding: _chart_axis_axis_label_Padding2.default.value,
          stroke: _chart_axis_axis_label_stroke_Color2.default.value
        }),
        grid: {
          fill: _chart_axis_grid_Fill2.default.value,
          stroke: 'none',
          pointerEvents: _chart_axis_grid_PointerEvents2.default.value,
          strokeLinecap: STROKE_LINE_CAP,
          strokeLinejoin: STROKE_LINE_JOIN
        },
        ticks: {
          fill: _chart_axis_tick_Fill2.default.value,
          size: _chart_axis_tick_Size2.default.value,
          stroke: _chart_axis_tick_stroke_Color2.default.value,
          strokeLinecap: STROKE_LINE_CAP,
          strokeLinejoin: STROKE_LINE_JOIN,
          strokeWidth: _chart_axis_tick_Width2.default.value
        },
        tickLabels: _objectSpread({}, LABEL_PROPS, {
          fill: _chart_axis_tick_label_Fill2.default.value
        })
      }
    }),
    bar: _objectSpread({}, LAYOUT_PROPS, {
      barWidth: _chart_bar_Width2.default.value,
      style: {
        data: {
          fill: _chart_bar_data_Fill2.default.value,
          padding: _chart_bar_data_Padding2.default.value,
          stroke: _chart_bar_data_stroke2.default.value,
          strokeWidth: _chart_bar_data_stroke_Width2.default.value
        },
        labels: LABEL_PROPS
      }
    }),
    boxplot: _objectSpread({}, LAYOUT_PROPS, {
      style: {
        max: {
          padding: _chart_boxplot_max_Padding2.default.value,
          stroke: _chart_boxplot_max_stroke_Color2.default.value,
          strokeWidth: _chart_boxplot_max_stroke_Width2.default.value
        },
        maxLabels: LABEL_PROPS,
        median: {
          padding: _chart_boxplot_median_Padding2.default.value,
          stroke: _chart_boxplot_median_stroke_Color2.default.value,
          strokeWidth: _chart_boxplot_median_stroke_Width2.default.value
        },
        medianLabels: LABEL_PROPS,
        min: {
          padding: _chart_boxplot_min_Padding2.default.value,
          stroke: _chart_boxplot_min_stroke_Color2.default.value,
          strokeWidth: _chart_boxplot_min_stroke_Width2.default.value
        },
        minLabels: LABEL_PROPS,
        q1: {
          fill: _chart_boxplot_lower_quartile_Fill2.default.value,
          padding: _chart_boxplot_lower_quartile_Padding2.default.value
        },
        q1Labels: LABEL_PROPS,
        q3: {
          fill: _chart_boxplot_upper_quartile_Fill2.default.value,
          padding: _chart_boxplot_upper_quartile_Padding2.default.value
        },
        q3Labels: LABEL_PROPS
      },
      boxWidth: _chart_boxplot_box_Width2.default.value
    }),
    candlestick: _objectSpread({}, LAYOUT_PROPS, {
      candleColors: {
        positive: _chart_candelstick_candle_positive_Color2.default.value,
        negative: _chart_candelstick_candle_negative_Color2.default.value
      },
      style: {
        data: {
          stroke: _chart_candelstick_data_stroke_Color2.default.value,
          strokeWidth: _chart_candelstick_data_stroke_Width2.default.value
        },
        labels: LABEL_CENTERED_PROPS
      }
    }),
    chart: _objectSpread({}, LAYOUT_PROPS),
    errorbar: _objectSpread({}, LAYOUT_PROPS, {
      borderWidth: _chart_errorbar_BorderWidth2.default.value,
      style: {
        data: {
          fill: _chart_errorbar_data_Fill2.default.value,
          opacity: _chart_errorbar_data_Opacity2.default.value,
          stroke: _chart_errorbar_data_stroke_Color2.default.value,
          strokeWidth: _chart_errorbar_data_stroke_Width2.default.value
        },
        labels: LABEL_CENTERED_PROPS
      }
    }),
    group: _objectSpread({}, LAYOUT_PROPS),
    legend: {
      gutter: _chart_legend_gutter_Width2.default.value,
      orientation: _chart_legend_orientation2.default.value,
      titleOrientation: _chart_legend_title_orientation2.default.value,
      style: {
        data: {
          type: _chart_legend_data_type2.default.value
        },
        labels: LABEL_PROPS,
        title: _objectSpread({}, LABEL_PROPS, {
          fontSize: TYPOGRAPHY_FONT_SIZE,
          padding: _chart_legend_title_Padding2.default.value
        })
      }
    },
    line: _objectSpread({}, LAYOUT_PROPS, {
      style: {
        data: {
          fill: _chart_line_data_Fill2.default.value,
          opacity: _chart_line_data_Opacity2.default.value,
          stroke: _chart_line_data_stroke_Color2.default.value,
          strokeWidth: _chart_line_data_stroke_Width2.default.value
        },
        labels: LABEL_CENTERED_PROPS
      }
    }),
    pie: {
      padding: _chart_pie_Padding2.default.value,
      style: {
        data: {
          padding: _chart_pie_data_Padding2.default.value,
          stroke: _chart_pie_data_stroke_Color2.default.value,
          strokeWidth: _chart_pie_data_stroke_Width2.default.value
        },
        labels: _objectSpread({}, LABEL_PROPS, {
          padding: _chart_pie_labels_Padding2.default.value
        })
      },
      height: _chart_pie_Height2.default.value,
      width: _chart_pie_Width2.default.value
    },
    scatter: _objectSpread({}, LAYOUT_PROPS, {
      style: {
        data: {
          fill: _chart_scatter_data_Fill2.default.value,
          opacity: _chart_scatter_data_Opacity2.default.value,
          stroke: _chart_scatter_data_stroke_Color2.default.value,
          strokeWidth: _chart_scatter_data_stroke_Width2.default.value
        },
        labels: LABEL_CENTERED_PROPS
      }
    }),
    stack: _objectSpread({}, LAYOUT_PROPS, {
      style: {
        data: {
          strokeWidth: _chart_stack_data_stroke_Width2.default.value
        }
      }
    }),
    tooltip: {
      cornerRadius: _chart_tooltip_corner_radius2.default.value,
      flyoutStyle: {
        cornerRadius: _chart_tooltip_flyoutStyle_corner_radius2.default.value,
        fill: _chart_tooltip_flyoutStyle_Fill2.default.value,
        // background
        pointerEvents: _chart_tooltip_flyoutStyle_PointerEvents2.default.value,
        stroke: _chart_tooltip_flyoutStyle_stroke_Color2.default.value,
        // border
        strokeWidth: _chart_tooltip_flyoutStyle_stroke_Width2.default.value
      },
      pointerLength: _chart_tooltip_pointer_length2.default.value,
      pointerWidth: _chart_tooltip_pointer_Width2.default.value,
      style: {
        fill: _chart_tooltip_Fill2.default.value,
        // text
        padding: _chart_tooltip_Padding2.default.value,
        pointerEvents: _chart_tooltip_PointerEvents2.default.value
      }
    },
    voronoi: _objectSpread({}, LAYOUT_PROPS, {
      style: {
        data: {
          fill: _chart_voronoi_data_Fill2.default.value,
          stroke: _chart_voronoi_data_stroke_Color2.default.value,
          strokeWidth: _chart_voronoi_data_stroke_Width2.default.value
        },
        labels: _objectSpread({}, LABEL_CENTERED_PROPS, {
          fill: _chart_voronoi_labels_Fill2.default.value,
          // text
          padding: _chart_voronoi_labels_Padding2.default.value,
          pointerEvents: _chart_voronoi_labels_PointerEvents2.default.value
        }),
        // Note: These properties override tooltip
        flyout: {
          fill: _chart_voronoi_flyout_stroke_Fill2.default.value,
          // background
          pointerEvents: _chart_voronoi_flyout_PointerEvents2.default.value,
          stroke: _chart_voronoi_flyout_stroke_Color2.default.value,
          // border
          strokeWidth: _chart_voronoi_flyout_stroke_Width2.default.value
        }
      }
    })
  };
});
