// @patternfly/react-charts/dist/umd/components/ChartTheme/themes/bullet-theme.js
(function (global, factory) {
  if (typeof define === "function" && define.amd) {
    define(["exports", "@patternfly/react-tokens/dist/js/chart_bullet_Height", "@patternfly/react-tokens/dist/js/chart_bullet_comparative_measure_Fill_Color", "@patternfly/react-tokens/dist/js/chart_bullet_comparative_measure_stroke_Color", "@patternfly/react-tokens/dist/js/chart_bullet_comparative_measure_stroke_Width", "@patternfly/react-tokens/dist/js/chart_bullet_comparative_measure_error_Fill_Color", "@patternfly/react-tokens/dist/js/chart_bullet_comparative_measure_error_stroke_Color", "@patternfly/react-tokens/dist/js/chart_bullet_comparative_measure_error_stroke_Width", "@patternfly/react-tokens/dist/js/chart_bullet_comparative_measure_warning_Fill_Color", "@patternfly/react-tokens/dist/js/chart_bullet_comparative_measure_warning_stroke_Color", "@patternfly/react-tokens/dist/js/chart_bullet_comparative_measure_warning_stroke_Width", "@patternfly/react-tokens/dist/js/chart_bullet_group_title_divider_Fill_Color", "@patternfly/react-tokens/dist/js/chart_bullet_group_title_divider_stroke_Color", "@patternfly/react-tokens/dist/js/chart_bullet_group_title_divider_stroke_Width", "@patternfly/react-tokens/dist/js/chart_color_black_100", "@patternfly/react-tokens/dist/js/chart_color_black_200", "@patternfly/react-tokens/dist/js/chart_color_black_300", "@patternfly/react-tokens/dist/js/chart_color_black_400", "@patternfly/react-tokens/dist/js/chart_color_black_500", "@patternfly/react-tokens/dist/js/chart_color_red_100", "@patternfly/react-tokens/dist/js/chart_color_red_200", "@patternfly/react-tokens/dist/js/chart_color_red_300", "@patternfly/react-tokens/dist/js/chart_color_red_400", "@patternfly/react-tokens/dist/js/chart_color_red_500", "@patternfly/react-tokens/dist/js/chart_global_layout_Padding"], factory);
  } else if (typeof exports !== "undefined") {
    factory(exports, require("@patternfly/react-tokens/dist/js/chart_bullet_Height"), require("@patternfly/react-tokens/dist/js/chart_bullet_comparative_measure_Fill_Color"), require("@patternfly/react-tokens/dist/js/chart_bullet_comparative_measure_stroke_Color"), require("@patternfly/react-tokens/dist/js/chart_bullet_comparative_measure_stroke_Width"), require("@patternfly/react-tokens/dist/js/chart_bullet_comparative_measure_error_Fill_Color"), require("@patternfly/react-tokens/dist/js/chart_bullet_comparative_measure_error_stroke_Color"), require("@patternfly/react-tokens/dist/js/chart_bullet_comparative_measure_error_stroke_Width"), require("@patternfly/react-tokens/dist/js/chart_bullet_comparative_measure_warning_Fill_Color"), require("@patternfly/react-tokens/dist/js/chart_bullet_comparative_measure_warning_stroke_Color"), require("@patternfly/react-tokens/dist/js/chart_bullet_comparative_measure_warning_stroke_Width"), require("@patternfly/react-tokens/dist/js/chart_bullet_group_title_divider_Fill_Color"), require("@patternfly/react-tokens/dist/js/chart_bullet_group_title_divider_stroke_Color"), require("@patternfly/react-tokens/dist/js/chart_bullet_group_title_divider_stroke_Width"), require("@patternfly/react-tokens/dist/js/chart_color_black_100"), require("@patternfly/react-tokens/dist/js/chart_color_black_200"), require("@patternfly/react-tokens/dist/js/chart_color_black_300"), require("@patternfly/react-tokens/dist/js/chart_color_black_400"), require("@patternfly/react-tokens/dist/js/chart_color_black_500"), require("@patternfly/react-tokens/dist/js/chart_color_red_100"), require("@patternfly/react-tokens/dist/js/chart_color_red_200"), require("@patternfly/react-tokens/dist/js/chart_color_red_300"), require("@patternfly/react-tokens/dist/js/chart_color_red_400"), require("@patternfly/react-tokens/dist/js/chart_color_red_500"), require("@patternfly/react-tokens/dist/js/chart_global_layout_Padding"));
  } else {
    var mod = {
      exports: {}
    };
    factory(mod.exports, global.chart_bullet_Height, global.chart_bullet_comparative_measure_Fill_Color, global.chart_bullet_comparative_measure_stroke_Color, global.chart_bullet_comparative_measure_stroke_Width, global.chart_bullet_comparative_measure_error_Fill_Color, global.chart_bullet_comparative_measure_error_stroke_Color, global.chart_bullet_comparative_measure_error_stroke_Width, global.chart_bullet_comparative_measure_warning_Fill_Color, global.chart_bullet_comparative_measure_warning_stroke_Color, global.chart_bullet_comparative_measure_warning_stroke_Width, global.chart_bullet_group_title_divider_Fill_Color, global.chart_bullet_group_title_divider_stroke_Color, global.chart_bullet_group_title_divider_stroke_Width, global.chart_color_black_100, global.chart_color_black_200, global.chart_color_black_300, global.chart_color_black_400, global.chart_color_black_500, global.chart_color_red_100, global.chart_color_red_200, global.chart_color_red_300, global.chart_color_red_400, global.chart_color_red_500, global.chart_global_layout_Padding);
    global.undefined = mod.exports;
  }
})(this, function (exports, _chart_bullet_Height, _chart_bullet_comparative_measure_Fill_Color, _chart_bullet_comparative_measure_stroke_Color, _chart_bullet_comparative_measure_stroke_Width, _chart_bullet_comparative_measure_error_Fill_Color, _chart_bullet_comparative_measure_error_stroke_Color, _chart_bullet_comparative_measure_error_stroke_Width, _chart_bullet_comparative_measure_warning_Fill_Color, _chart_bullet_comparative_measure_warning_stroke_Color, _chart_bullet_comparative_measure_warning_stroke_Width, _chart_bullet_group_title_divider_Fill_Color, _chart_bullet_group_title_divider_stroke_Color, _chart_bullet_group_title_divider_stroke_Width, _chart_color_black_, _chart_color_black_3, _chart_color_black_5, _chart_color_black_7, _chart_color_black_9, _chart_color_red_, _chart_color_red_3, _chart_color_red_5, _chart_color_red_7, _chart_color_red_9, _chart_global_layout_Padding) {
  "use strict";

  Object.defineProperty(exports, "__esModule", {
    value: true
  });
  exports.BulletQualitativeRangeTheme = exports.BulletPrimarySegmentedMeasureTheme = exports.BulletPrimaryNegativeMeasureTheme = exports.BulletPrimaryDotMeasureTheme = exports.BulletGroupTitleTheme = exports.BulletComparativeWarningMeasureTheme = exports.BulletComparativeMeasureTheme = exports.BulletComparativeErrorMeasureTheme = exports.BulletTheme = undefined;

  var _chart_bullet_Height2 = _interopRequireDefault(_chart_bullet_Height);

  var _chart_bullet_comparative_measure_Fill_Color2 = _interopRequireDefault(_chart_bullet_comparative_measure_Fill_Color);

  var _chart_bullet_comparative_measure_stroke_Color2 = _interopRequireDefault(_chart_bullet_comparative_measure_stroke_Color);

  var _chart_bullet_comparative_measure_stroke_Width2 = _interopRequireDefault(_chart_bullet_comparative_measure_stroke_Width);

  var _chart_bullet_comparative_measure_error_Fill_Color2 = _interopRequireDefault(_chart_bullet_comparative_measure_error_Fill_Color);

  var _chart_bullet_comparative_measure_error_stroke_Color2 = _interopRequireDefault(_chart_bullet_comparative_measure_error_stroke_Color);

  var _chart_bullet_comparative_measure_error_stroke_Width2 = _interopRequireDefault(_chart_bullet_comparative_measure_error_stroke_Width);

  var _chart_bullet_comparative_measure_warning_Fill_Color2 = _interopRequireDefault(_chart_bullet_comparative_measure_warning_Fill_Color);

  var _chart_bullet_comparative_measure_warning_stroke_Color2 = _interopRequireDefault(_chart_bullet_comparative_measure_warning_stroke_Color);

  var _chart_bullet_comparative_measure_warning_stroke_Width2 = _interopRequireDefault(_chart_bullet_comparative_measure_warning_stroke_Width);

  var _chart_bullet_group_title_divider_Fill_Color2 = _interopRequireDefault(_chart_bullet_group_title_divider_Fill_Color);

  var _chart_bullet_group_title_divider_stroke_Color2 = _interopRequireDefault(_chart_bullet_group_title_divider_stroke_Color);

  var _chart_bullet_group_title_divider_stroke_Width2 = _interopRequireDefault(_chart_bullet_group_title_divider_stroke_Width);

  var _chart_color_black_2 = _interopRequireDefault(_chart_color_black_);

  var _chart_color_black_4 = _interopRequireDefault(_chart_color_black_3);

  var _chart_color_black_6 = _interopRequireDefault(_chart_color_black_5);

  var _chart_color_black_8 = _interopRequireDefault(_chart_color_black_7);

  var _chart_color_black_10 = _interopRequireDefault(_chart_color_black_9);

  var _chart_color_red_2 = _interopRequireDefault(_chart_color_red_);

  var _chart_color_red_4 = _interopRequireDefault(_chart_color_red_3);

  var _chart_color_red_6 = _interopRequireDefault(_chart_color_red_5);

  var _chart_color_red_8 = _interopRequireDefault(_chart_color_red_7);

  var _chart_color_red_10 = _interopRequireDefault(_chart_color_red_9);

  var _chart_global_layout_Padding2 = _interopRequireDefault(_chart_global_layout_Padding);

  function _interopRequireDefault(obj) {
    return obj && obj.__esModule ? obj : {
      default: obj
    };
  }

  /* eslint-disable camelcase */
  // See https://docs.google.com/document/d/1cw10pJFXWruB1SA8TQwituxn5Ss6KpxYPCOYGrH8qAY/edit#
  // See https://www.patternfly.org/v3/pattern-library/data-visualization/bullet-chart/#design
  // Bullet theme
  const BulletTheme = exports.BulletTheme = {
    chart: {
      height: _chart_bullet_Height2.default.value
    }
  }; // Bullet comparative measure error theme

  const BulletComparativeErrorMeasureTheme = exports.BulletComparativeErrorMeasureTheme = {
    bar: {
      height: _chart_bullet_Height2.default.value,
      style: {
        data: {
          fill: _chart_bullet_comparative_measure_error_Fill_Color2.default.value,
          stroke: _chart_bullet_comparative_measure_error_stroke_Color2.default.value,
          strokeWidth: _chart_bullet_comparative_measure_error_stroke_Width2.default.value
        }
      }
    }
  }; // Bullet comparative measure theme

  const BulletComparativeMeasureTheme = exports.BulletComparativeMeasureTheme = {
    bar: {
      height: _chart_bullet_Height2.default.value,
      style: {
        data: {
          fill: _chart_bullet_comparative_measure_Fill_Color2.default.value,
          stroke: _chart_bullet_comparative_measure_stroke_Color2.default.value,
          strokeWidth: _chart_bullet_comparative_measure_stroke_Width2.default.value
        }
      }
    }
  }; // Bullet comparative measure warning theme

  const BulletComparativeWarningMeasureTheme = exports.BulletComparativeWarningMeasureTheme = {
    bar: {
      height: _chart_bullet_Height2.default.value,
      style: {
        data: {
          fill: _chart_bullet_comparative_measure_warning_Fill_Color2.default.value,
          stroke: _chart_bullet_comparative_measure_warning_stroke_Color2.default.value,
          strokeWidth: _chart_bullet_comparative_measure_warning_stroke_Width2.default.value
        }
      }
    }
  }; // Bullet group title theme

  const BulletGroupTitleTheme = exports.BulletGroupTitleTheme = {
    chart: {
      padding: {
        bottom: 0,
        left: 0,
        right: 0,
        top: _chart_global_layout_Padding2.default.value
      }
    },
    line: {
      style: {
        data: {
          fill: _chart_bullet_group_title_divider_Fill_Color2.default.value,
          stroke: _chart_bullet_group_title_divider_stroke_Color2.default.value,
          strokeWidth: _chart_bullet_group_title_divider_stroke_Width2.default.value
        }
      }
    }
  }; // Bullet primary dot measure theme

  const BulletPrimaryDotMeasureTheme = exports.BulletPrimaryDotMeasureTheme = {
    group: {
      height: _chart_bullet_Height2.default.value
    }
  }; // Bullet primary negative measure theme

  const BulletPrimaryNegativeMeasureTheme = exports.BulletPrimaryNegativeMeasureTheme = {
    group: {
      colorScale: [_chart_color_red_2.default.value, _chart_color_red_4.default.value, _chart_color_red_6.default.value, _chart_color_red_8.default.value, _chart_color_red_10.default.value],
      height: _chart_bullet_Height2.default.value
    }
  }; // Bullet primary segmented measure theme

  const BulletPrimarySegmentedMeasureTheme = exports.BulletPrimarySegmentedMeasureTheme = {
    group: {
      height: _chart_bullet_Height2.default.value
    }
  }; // Bullet qualitative range theme

  const BulletQualitativeRangeTheme = exports.BulletQualitativeRangeTheme = {
    group: {
      colorScale: [_chart_color_black_2.default.value, _chart_color_black_4.default.value, _chart_color_black_6.default.value, _chart_color_black_8.default.value, _chart_color_black_10.default.value],
      height: _chart_bullet_Height2.default.value
    }
  };
});
