// @patternfly/react-charts/dist/umd/components/ChartTheme/themes/donut-threshold-theme.js
(function (global, factory) {
  if (typeof define === "function" && define.amd) {
    define(["exports", "@patternfly/react-tokens/dist/js/chart_donut_threshold_first_Color", "@patternfly/react-tokens/dist/js/chart_donut_threshold_second_Color", "@patternfly/react-tokens/dist/js/chart_donut_threshold_third_Color", "@patternfly/react-tokens/dist/js/chart_donut_threshold_dynamic_pie_Height", "@patternfly/react-tokens/dist/js/chart_donut_threshold_dynamic_pie_Padding", "@patternfly/react-tokens/dist/js/chart_donut_threshold_dynamic_pie_Width", "@patternfly/react-tokens/dist/js/chart_donut_threshold_static_pie_Height", "@patternfly/react-tokens/dist/js/chart_donut_threshold_static_pie_angle_Padding", "@patternfly/react-tokens/dist/js/chart_donut_threshold_static_pie_Padding", "@patternfly/react-tokens/dist/js/chart_donut_threshold_static_pie_Width"], factory);
  } else if (typeof exports !== "undefined") {
    factory(exports, require("@patternfly/react-tokens/dist/js/chart_donut_threshold_first_Color"), require("@patternfly/react-tokens/dist/js/chart_donut_threshold_second_Color"), require("@patternfly/react-tokens/dist/js/chart_donut_threshold_third_Color"), require("@patternfly/react-tokens/dist/js/chart_donut_threshold_dynamic_pie_Height"), require("@patternfly/react-tokens/dist/js/chart_donut_threshold_dynamic_pie_Padding"), require("@patternfly/react-tokens/dist/js/chart_donut_threshold_dynamic_pie_Width"), require("@patternfly/react-tokens/dist/js/chart_donut_threshold_static_pie_Height"), require("@patternfly/react-tokens/dist/js/chart_donut_threshold_static_pie_angle_Padding"), require("@patternfly/react-tokens/dist/js/chart_donut_threshold_static_pie_Padding"), require("@patternfly/react-tokens/dist/js/chart_donut_threshold_static_pie_Width"));
  } else {
    var mod = {
      exports: {}
    };
    factory(mod.exports, global.chart_donut_threshold_first_Color, global.chart_donut_threshold_second_Color, global.chart_donut_threshold_third_Color, global.chart_donut_threshold_dynamic_pie_Height, global.chart_donut_threshold_dynamic_pie_Padding, global.chart_donut_threshold_dynamic_pie_Width, global.chart_donut_threshold_static_pie_Height, global.chart_donut_threshold_static_pie_angle_Padding, global.chart_donut_threshold_static_pie_Padding, global.chart_donut_threshold_static_pie_Width);
    global.undefined = mod.exports;
  }
})(this, function (exports, _chart_donut_threshold_first_Color, _chart_donut_threshold_second_Color, _chart_donut_threshold_third_Color, _chart_donut_threshold_dynamic_pie_Height, _chart_donut_threshold_dynamic_pie_Padding, _chart_donut_threshold_dynamic_pie_Width, _chart_donut_threshold_static_pie_Height, _chart_donut_threshold_static_pie_angle_Padding, _chart_donut_threshold_static_pie_Padding, _chart_donut_threshold_static_pie_Width) {
  "use strict";

  Object.defineProperty(exports, "__esModule", {
    value: true
  });
  exports.DonutThresholdStaticTheme = exports.DonutThresholdDynamicTheme = undefined;

  var _chart_donut_threshold_first_Color2 = _interopRequireDefault(_chart_donut_threshold_first_Color);

  var _chart_donut_threshold_second_Color2 = _interopRequireDefault(_chart_donut_threshold_second_Color);

  var _chart_donut_threshold_third_Color2 = _interopRequireDefault(_chart_donut_threshold_third_Color);

  var _chart_donut_threshold_dynamic_pie_Height2 = _interopRequireDefault(_chart_donut_threshold_dynamic_pie_Height);

  var _chart_donut_threshold_dynamic_pie_Padding2 = _interopRequireDefault(_chart_donut_threshold_dynamic_pie_Padding);

  var _chart_donut_threshold_dynamic_pie_Width2 = _interopRequireDefault(_chart_donut_threshold_dynamic_pie_Width);

  var _chart_donut_threshold_static_pie_Height2 = _interopRequireDefault(_chart_donut_threshold_static_pie_Height);

  var _chart_donut_threshold_static_pie_angle_Padding2 = _interopRequireDefault(_chart_donut_threshold_static_pie_angle_Padding);

  var _chart_donut_threshold_static_pie_Padding2 = _interopRequireDefault(_chart_donut_threshold_static_pie_Padding);

  var _chart_donut_threshold_static_pie_Width2 = _interopRequireDefault(_chart_donut_threshold_static_pie_Width);

  function _interopRequireDefault(obj) {
    return obj && obj.__esModule ? obj : {
      default: obj
    };
  }

  /* eslint-disable camelcase */
  // Donut threshold dynamic theme
  const DonutThresholdDynamicTheme = exports.DonutThresholdDynamicTheme = {
    legend: {
      colorScale: [_chart_donut_threshold_second_Color2.default.value, _chart_donut_threshold_third_Color2.default.value]
    },
    pie: {
      height: _chart_donut_threshold_dynamic_pie_Height2.default.value,
      padding: _chart_donut_threshold_dynamic_pie_Padding2.default.value,
      width: _chart_donut_threshold_dynamic_pie_Width2.default.value
    }
  }; // Donut threshold static theme

  const DonutThresholdStaticTheme = exports.DonutThresholdStaticTheme = {
    pie: {
      colorScale: [_chart_donut_threshold_first_Color2.default.value, _chart_donut_threshold_second_Color2.default.value, _chart_donut_threshold_third_Color2.default.value],
      height: _chart_donut_threshold_static_pie_Height2.default.value,
      padAngle: _chart_donut_threshold_static_pie_angle_Padding2.default.value,
      padding: _chart_donut_threshold_static_pie_Padding2.default.value,
      width: _chart_donut_threshold_static_pie_Width2.default.value
    }
  };
});
