// @patternfly/react-charts/dist/umd/components/ChartTheme/themes/threshold-theme.js
(function (global, factory) {
  if (typeof define === "function" && define.amd) {
    define(["exports", "@patternfly/react-tokens/dist/js/chart_threshold_stroke_dash_array", "@patternfly/react-tokens/dist/js/chart_threshold_stroke_Width"], factory);
  } else if (typeof exports !== "undefined") {
    factory(exports, require("@patternfly/react-tokens/dist/js/chart_threshold_stroke_dash_array"), require("@patternfly/react-tokens/dist/js/chart_threshold_stroke_Width"));
  } else {
    var mod = {
      exports: {}
    };
    factory(mod.exports, global.chart_threshold_stroke_dash_array, global.chart_threshold_stroke_Width);
    global.undefined = mod.exports;
  }
})(this, function (exports, _chart_threshold_stroke_dash_array, _chart_threshold_stroke_Width) {
  "use strict";

  Object.defineProperty(exports, "__esModule", {
    value: true
  });
  exports.ThresholdTheme = undefined;

  var _chart_threshold_stroke_dash_array2 = _interopRequireDefault(_chart_threshold_stroke_dash_array);

  var _chart_threshold_stroke_Width2 = _interopRequireDefault(_chart_threshold_stroke_Width);

  function _interopRequireDefault(obj) {
    return obj && obj.__esModule ? obj : {
      default: obj
    };
  }

  /* eslint-disable camelcase */
  // Threshold theme
  const ThresholdTheme = exports.ThresholdTheme = {
    line: {
      style: {
        data: {
          strokeDasharray: _chart_threshold_stroke_dash_array2.default.value,
          strokeWidth: _chart_threshold_stroke_Width2.default.value
        }
      }
    }
  };
});
