// @patternfly/react-charts/dist/js/components/ChartTheme/styles/donut-styles.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.DonutStyles = void 0;

var _chart_global_FontSize_sm = _interopRequireDefault(require("@patternfly/react-tokens/dist/js/chart_global_FontSize_sm"));

var _chart_global_FontSize_2xl = _interopRequireDefault(require("@patternfly/react-tokens/dist/js/chart_global_FontSize_2xl"));

var _chart_donut_label_subtitle_Fill = _interopRequireDefault(require("@patternfly/react-tokens/dist/js/chart_donut_label_subtitle_Fill"));

var _chart_donut_label_subtitle_position = _interopRequireDefault(require("@patternfly/react-tokens/dist/js/chart_donut_label_subtitle_position"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

/* eslint-disable camelcase */
// Donut styles
var DonutStyles = {
  label: {
    subTitle: {
      // Victory props only
      fill: _chart_donut_label_subtitle_Fill["default"].value,
      fontSize: _chart_global_FontSize_sm["default"].value
    },
    subTitlePosition: _chart_donut_label_subtitle_position["default"].value,
    title: {
      // Victory props only
      fontSize: _chart_global_FontSize_2xl["default"].value
    }
  }
};
exports.DonutStyles = DonutStyles;
