// @patternfly/react-charts/dist/umd/components/ChartTheme/styles/bullet-styles.js
(function (global, factory) {
  if (typeof define === "function" && define.amd) {
    define(["exports", "@patternfly/react-tokens/dist/js/chart_bullet_axis_tick_count", "@patternfly/react-tokens/dist/js/chart_bullet_comparative_measure_Width", "@patternfly/react-tokens/dist/js/chart_bullet_comparative_measure_error_Width", "@patternfly/react-tokens/dist/js/chart_bullet_comparative_measure_warning_Width", "@patternfly/react-tokens/dist/js/chart_bullet_label_subtitle_Fill", "@patternfly/react-tokens/dist/js/chart_bullet_primary_measure_dot_size", "@patternfly/react-tokens/dist/js/chart_bullet_primary_measure_segmented_Width", "@patternfly/react-tokens/dist/js/chart_bullet_qualitative_range_Width", "@patternfly/react-tokens/dist/js/chart_global_FontSize_2xl", "@patternfly/react-tokens/dist/js/chart_global_FontSize_sm", "@patternfly/react-tokens/dist/js/chart_global_FontSize_lg"], factory);
  } else if (typeof exports !== "undefined") {
    factory(exports, require("@patternfly/react-tokens/dist/js/chart_bullet_axis_tick_count"), require("@patternfly/react-tokens/dist/js/chart_bullet_comparative_measure_Width"), require("@patternfly/react-tokens/dist/js/chart_bullet_comparative_measure_error_Width"), require("@patternfly/react-tokens/dist/js/chart_bullet_comparative_measure_warning_Width"), require("@patternfly/react-tokens/dist/js/chart_bullet_label_subtitle_Fill"), require("@patternfly/react-tokens/dist/js/chart_bullet_primary_measure_dot_size"), require("@patternfly/react-tokens/dist/js/chart_bullet_primary_measure_segmented_Width"), require("@patternfly/react-tokens/dist/js/chart_bullet_qualitative_range_Width"), require("@patternfly/react-tokens/dist/js/chart_global_FontSize_2xl"), require("@patternfly/react-tokens/dist/js/chart_global_FontSize_sm"), require("@patternfly/react-tokens/dist/js/chart_global_FontSize_lg"));
  } else {
    var mod = {
      exports: {}
    };
    factory(mod.exports, global.chart_bullet_axis_tick_count, global.chart_bullet_comparative_measure_Width, global.chart_bullet_comparative_measure_error_Width, global.chart_bullet_comparative_measure_warning_Width, global.chart_bullet_label_subtitle_Fill, global.chart_bullet_primary_measure_dot_size, global.chart_bullet_primary_measure_segmented_Width, global.chart_bullet_qualitative_range_Width, global.chart_global_FontSize_2xl, global.chart_global_FontSize_sm, global.chart_global_FontSize_lg);
    global.undefined = mod.exports;
  }
})(this, function (exports, _chart_bullet_axis_tick_count, _chart_bullet_comparative_measure_Width, _chart_bullet_comparative_measure_error_Width, _chart_bullet_comparative_measure_warning_Width, _chart_bullet_label_subtitle_Fill, _chart_bullet_primary_measure_dot_size, _chart_bullet_primary_measure_segmented_Width, _chart_bullet_qualitative_range_Width, _chart_global_FontSize_2xl, _chart_global_FontSize_sm, _chart_global_FontSize_lg) {
  "use strict";

  Object.defineProperty(exports, "__esModule", {
    value: true
  });
  exports.BulletStyles = undefined;

  var _chart_bullet_axis_tick_count2 = _interopRequireDefault(_chart_bullet_axis_tick_count);

  var _chart_bullet_comparative_measure_Width2 = _interopRequireDefault(_chart_bullet_comparative_measure_Width);

  var _chart_bullet_comparative_measure_error_Width2 = _interopRequireDefault(_chart_bullet_comparative_measure_error_Width);

  var _chart_bullet_comparative_measure_warning_Width2 = _interopRequireDefault(_chart_bullet_comparative_measure_warning_Width);

  var _chart_bullet_label_subtitle_Fill2 = _interopRequireDefault(_chart_bullet_label_subtitle_Fill);

  var _chart_bullet_primary_measure_dot_size2 = _interopRequireDefault(_chart_bullet_primary_measure_dot_size);

  var _chart_bullet_primary_measure_segmented_Width2 = _interopRequireDefault(_chart_bullet_primary_measure_segmented_Width);

  var _chart_bullet_qualitative_range_Width2 = _interopRequireDefault(_chart_bullet_qualitative_range_Width);

  var _chart_global_FontSize_2xl2 = _interopRequireDefault(_chart_global_FontSize_2xl);

  var _chart_global_FontSize_sm2 = _interopRequireDefault(_chart_global_FontSize_sm);

  var _chart_global_FontSize_lg2 = _interopRequireDefault(_chart_global_FontSize_lg);

  function _interopRequireDefault(obj) {
    return obj && obj.__esModule ? obj : {
      default: obj
    };
  }

  /* eslint-disable camelcase */
  // Bullet styles
  const BulletStyles = exports.BulletStyles = {
    axisTickCount: _chart_bullet_axis_tick_count2.default.value,
    comparativeMeasureErrorWidth: _chart_bullet_comparative_measure_error_Width2.default.value,
    comparativeMeasureWarningWidth: _chart_bullet_comparative_measure_warning_Width2.default.value,
    comparativeMeasureWidth: _chart_bullet_comparative_measure_Width2.default.value,
    label: {
      groupTitle: {
        // Victory props only
        fontSize: _chart_global_FontSize_2xl2.default.value
      },
      subTitle: {
        // Victory props only
        fill: _chart_bullet_label_subtitle_Fill2.default.value,
        fontSize: _chart_global_FontSize_sm2.default.value
      },
      title: {
        // Victory props only
        fontSize: _chart_global_FontSize_lg2.default.value
      }
    },
    primaryDotMeasureSize: _chart_bullet_primary_measure_dot_size2.default.value,
    primarySegmentedMeasureWidth: _chart_bullet_primary_measure_segmented_Width2.default.value,
    qualitativeRangeWidth: _chart_bullet_qualitative_range_Width2.default.value
  };
});
