// @patternfly/react-charts/dist/umd/components/ChartTheme/styles/common-styles.js
(function (global, factory) {
  if (typeof define === "function" && define.amd) {
    define(["exports", "@patternfly/react-tokens/dist/js/chart_global_FontFamily", "@patternfly/react-tokens/dist/js/chart_global_FontSize_sm", "@patternfly/react-tokens/dist/js/chart_global_label_Margin", "@patternfly/react-tokens/dist/js/chart_global_letter_spacing", "@patternfly/react-tokens/dist/js/chart_legend_Margin", "@patternfly/react-tokens/dist/js/chart_legend_position"], factory);
  } else if (typeof exports !== "undefined") {
    factory(exports, require("@patternfly/react-tokens/dist/js/chart_global_FontFamily"), require("@patternfly/react-tokens/dist/js/chart_global_FontSize_sm"), require("@patternfly/react-tokens/dist/js/chart_global_label_Margin"), require("@patternfly/react-tokens/dist/js/chart_global_letter_spacing"), require("@patternfly/react-tokens/dist/js/chart_legend_Margin"), require("@patternfly/react-tokens/dist/js/chart_legend_position"));
  } else {
    var mod = {
      exports: {}
    };
    factory(mod.exports, global.chart_global_FontFamily, global.chart_global_FontSize_sm, global.chart_global_label_Margin, global.chart_global_letter_spacing, global.chart_legend_Margin, global.chart_legend_position);
    global.undefined = mod.exports;
  }
})(this, function (exports, _chart_global_FontFamily, _chart_global_FontSize_sm, _chart_global_label_Margin, _chart_global_letter_spacing, _chart_legend_Margin, _chart_legend_position) {
  "use strict";

  Object.defineProperty(exports, "__esModule", {
    value: true
  });
  exports.CommonStyles = undefined;

  var _chart_global_FontFamily2 = _interopRequireDefault(_chart_global_FontFamily);

  var _chart_global_FontSize_sm2 = _interopRequireDefault(_chart_global_FontSize_sm);

  var _chart_global_label_Margin2 = _interopRequireDefault(_chart_global_label_Margin);

  var _chart_global_letter_spacing2 = _interopRequireDefault(_chart_global_letter_spacing);

  var _chart_legend_Margin2 = _interopRequireDefault(_chart_legend_Margin);

  var _chart_legend_position2 = _interopRequireDefault(_chart_legend_position);

  function _interopRequireDefault(obj) {
    return obj && obj.__esModule ? obj : {
      default: obj
    };
  }

  /* eslint-disable camelcase */
  // Typography
  const TYPOGRAPHY_FONT_FAMILY = _chart_global_FontFamily2.default.var;
  const TYPOGRAPHY_LETTER_SPACING = _chart_global_letter_spacing2.default.var;
  const TYPOGRAPHY_FONT_SIZE = _chart_global_FontSize_sm2.default.value;
  const CommonStyles = exports.CommonStyles = {
    label: {
      fontFamily: TYPOGRAPHY_FONT_FAMILY,
      fontSize: TYPOGRAPHY_FONT_SIZE,
      letterSpacing: TYPOGRAPHY_LETTER_SPACING,
      margin: _chart_global_label_Margin2.default.value
    },
    legend: {
      margin: _chart_legend_Margin2.default.value,
      position: _chart_legend_position2.default.value
    }
  };
});
