// @patternfly/react-charts/dist/umd/components/ChartTheme/styles/donut-styles.js
(function (global, factory) {
  if (typeof define === "function" && define.amd) {
    define(["exports", "@patternfly/react-tokens/dist/js/chart_global_FontSize_sm", "@patternfly/react-tokens/dist/js/chart_global_FontSize_2xl", "@patternfly/react-tokens/dist/js/chart_donut_label_subtitle_Fill", "@patternfly/react-tokens/dist/js/chart_donut_label_subtitle_position"], factory);
  } else if (typeof exports !== "undefined") {
    factory(exports, require("@patternfly/react-tokens/dist/js/chart_global_FontSize_sm"), require("@patternfly/react-tokens/dist/js/chart_global_FontSize_2xl"), require("@patternfly/react-tokens/dist/js/chart_donut_label_subtitle_Fill"), require("@patternfly/react-tokens/dist/js/chart_donut_label_subtitle_position"));
  } else {
    var mod = {
      exports: {}
    };
    factory(mod.exports, global.chart_global_FontSize_sm, global.chart_global_FontSize_2xl, global.chart_donut_label_subtitle_Fill, global.chart_donut_label_subtitle_position);
    global.undefined = mod.exports;
  }
})(this, function (exports, _chart_global_FontSize_sm, _chart_global_FontSize_2xl, _chart_donut_label_subtitle_Fill, _chart_donut_label_subtitle_position) {
  "use strict";

  Object.defineProperty(exports, "__esModule", {
    value: true
  });
  exports.DonutStyles = undefined;

  var _chart_global_FontSize_sm2 = _interopRequireDefault(_chart_global_FontSize_sm);

  var _chart_global_FontSize_2xl2 = _interopRequireDefault(_chart_global_FontSize_2xl);

  var _chart_donut_label_subtitle_Fill2 = _interopRequireDefault(_chart_donut_label_subtitle_Fill);

  var _chart_donut_label_subtitle_position2 = _interopRequireDefault(_chart_donut_label_subtitle_position);

  function _interopRequireDefault(obj) {
    return obj && obj.__esModule ? obj : {
      default: obj
    };
  }

  /* eslint-disable camelcase */
  // Donut styles
  const DonutStyles = exports.DonutStyles = {
    label: {
      subTitle: {
        // Victory props only
        fill: _chart_donut_label_subtitle_Fill2.default.value,
        fontSize: _chart_global_FontSize_sm2.default.value
      },
      subTitlePosition: _chart_donut_label_subtitle_position2.default.value,
      title: {
        // Victory props only
        fontSize: _chart_global_FontSize_2xl2.default.value
      }
    }
  };
});
